APMrover2/GCS_Mavlink.cpp

```diff
@@ -1110,16 +1110,21 @@ void GCS_MAVLINK_Rover::handleMessage(mavlink_message_t* msg)
 
             float target_speed = 0.0f;
             float target_yaw_cd = 0.0f;
+            float target_dir = 1.0f;  // by default, we consider that the rover will use drive forward
 
             // consume velocity and convert to target speed and heading
             if (!vel_ignore) {
                 // convert vector length into a speed
                 target_speed = constrain_float(safe_sqrt(sq(packet.vx) + sq(packet.vy)), -rover.g.speed_cruise, rover.g.speed_cruise);
                 // convert vector direction to target yaw
                 target_yaw_cd = degrees(atan2f(packet.vy, packet.vx)) * 100.0f;
+
                 // rotate target yaw if provided in body-frame
                 if (packet.coordinate_frame == MAV_FRAME_BODY_NED || packet.coordinate_frame == MAV_FRAME_BODY_OFFSET_NED) {
                     target_yaw_cd = wrap_180_cd(target_yaw_cd + rover.ahrs.yaw_sensor);
+                    if (is_negative(packet.vx)) {
+                      target_dir = -1.0f;
```

Comment:
a minor thing but I noticed on this PR and another that the indentation is only 2 characters.  Maybe a default setting is off in your editor?

Reply:
arf ...Yes ... that because mavros is indent with 2 spaces ... I need to correct that